### PR TITLE
[Inventory] Remove backorders & available on demand support, add tracking toggle on Stockable

### PR DESCRIPTION
**Pull request by pamil**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | yes
| Related tickets | -
| License         | MIT

Temporarily removing backorders & available on demand support - will rewrite it soon (with scenarios :tada:), adding tracking toggle on stockable.

### CODE REVIEW

File: docs/book/inventory.rst [hunk 1]
```diff
@@ -52,9 +38,9 @@ Normally, inventory units are created automatically by Sylius and you do not nee
     use Sylius\Component\Inventory\Model\InventoryUnitInterface;
 
     $variant = // Get variant from product.
-    $inventoryUnits = $this->get('sylius.inventory_unit_factory')->create($variant, 6, InventoryUnitInterface::STATE_BACKORDER);
+    $inventoryUnits = $this->get('sylius.inventory_unit_factory')->create($variant, 6, InventoryUnitInterface::STATE_RETURNED);
```
Comment (michalmarcinkowski) [inline docs/book/inventory.rst]: We do not support returns out of the box, so the state should be different.

Btw. this state should be completely removed.

File: docs/bundles/SyliusInventoryBundle/models.rst [hunk 2]
```diff
@@ -6,12 +6,12 @@ Here is a quick reference for the default models.
 InventoryUnit
 -------------
 
-Each unit holds a reference to a stockable object and its state, which can be **sold** or **backordered**.
-It also provides some handy shortcut methods like `isSold`, `isBackordered` and `getSku`.
+Each unit holds a reference to a stockable object and its state, which can be **sold** or **returned**.
+It also provides some handy shortcut methods like `isSold` and `getSku`.
```
Comment (michalmarcinkowski) [inline docs/bundles/SyliusInventoryBundle/models.rst]: `getSku` is not a part of the interface anymore.

File: src/Sylius/Component/Inventory/composer.json [hunk 3]
```diff
@@ -20,9 +20,7 @@
         }
     ],
     "require": {
-        "php": "^5.6|^7.0",
-
-        "sylius/resource": "^1.0"
+        "php": "^5.6|^7.0"
```
Comment (tuka217) [inline src/Sylius/Component/Inventory/composer.json]: Is this change needed?

Comment (pamil) [inline src/Sylius/Component/Inventory/composer.json]: Yes, we don't need `sylius/resource` any more.

**Comment by pjedrzejewski** [comment]
Thanks Kamil! 👍
